atox/src/main/kotlin/ToxService.kt

```diff
@@ -124,6 +131,20 @@ class ToxService : LifecycleService() {
                 }
         }
 
+        lifecycleScope.launch(Dispatchers.Default) {
+            friendRequestManager.getAll()
+                .filterNotNull()
+                .flowWithLifecycle(lifecycle)
+                .collect { friendRequests ->
+                    val finishedFriendRequests = knownFriendRequests.minus(friendRequests.toSet())
+                    finishedFriendRequests.forEach {
+                        knownFriendRequests.remove(it)
+                        notifier.cancel(it.publicKey.hashCode())
+                    }
+                    knownFriendRequests.addAll(friendRequests.minus(finishedFriendRequests))
```

Comment:
`finishedFriendRequests` can never contain requests from `friendRequests` due to line 139 doing
```kotlin
val finishedFriendRequests = knownFriendRequests.minus(friendRequests.toSet())
```
so you can change this line to `knownFriendRequests.addAll(friendRequests)` or something along those lines, I think?

Can you please also squash your commits to just be 1? :P